addons/account_peppol/models/res_config_settings.py

```diff
@@ -240,46 +240,29 @@ def button_cancel_peppol_registration(self):
         Sets the peppol registration to canceled
         - If the user is active on the SMP, we can't just cancel it.
           They have to request a migration key using the `button_migrate_peppol_registration` action
+          or deregister.
         - 'not_registered', 'rejected', 'canceled' proxy states mean that canceling the registration
           makes no sense, so we don't do it
         - Calls the IAP server first before setting the state as canceled on the client side,
           in case they've been activated on the IAP side in the meantime
         """
         self.ensure_one()
+        # check if the participant has been already registered
+        self.env.ref('account_peppol.ir_cron_peppol_get_participant_status')._trigger()
```

Comment:
I don't like this part. You are forcing the cron for all users, not only the needed one. The CRON will be updated and you risk some access rights issues here. Instead, you should update the state only for the current company without calling the cron.